src/widget/wsearchlineedit.cpp

```diff
@@ -465,27 +473,31 @@ void WSearchLineEdit::slotTriggerSearch() {
 
 /// saves the current query as selection
 void WSearchLineEdit::slotSaveSearch() {
+    m_saveTimer.stop();
+    QString cText = currentText().trimmed();
+    int cIndex = findCurrentTextIndex();
```

Comment:
`trimmed()` must be used consistently. Currently, `findCurrentTextIndex()` does not apply it. There might be even more occasions I didn't spot.

Instead of inlining this post-processing everywhere it should better be enclosed in a function to prevent those inconsistencies -> DRY.

Reply:
Currently `trimmed()` is in only three rather isolated and independent places, thus outsourcing it to a separate function doesn't make much sense to me.
* save to config
* read from config (IMO actually only used when upgrading to this version)
* save search

`findCurrentTextIndex()` doesn't need to use the trimmed text. It is used to decide whether the to call `saveSearch()` which then checks if the trimmed text needs to be stored. In other places `findCurrentTextIndex()` is called _after_ the `saveSearch` already examined the text.